### ISSUE TITLE
Particles2D should have special position

### ISSUE THREAD

**Issue by mateusak** [godotengine/godot#15960]
You should be able to override the natural position inheritance of Particles2D. Imagine your player have a gun, it fires, and at the point of collision you want some particles.  You could achieve that in two ways:

1. Spawn the particles in some sort of particles controller node, so if the player shoots and moves at the same time it won't affect the particles.
2. Or, if somehow you could override the natural position inheritance, just make it a sub-node of the gun and set its position.

Actually, it would be nice to have such feature in all nodes, but its specially critical for particles.

**TL;DR**
**Implement a 'Disable position inheritance' checkbox for Particles2D nodes.**

_This has nothing to do with local coordinates on particles. I'm aware of that feature, but it doesn't relate to this issue._

**Comment by Zylann** [comment]
Why not spawn those particles outside of the player's hierarchy? That's what I've been doing for a long time in all engines.

**Comment by mateusak** [comment]
@Zylann Yeah, that's the first option I mentioned. See, let's assume you were to spawn it outside the player. Now imagine you have a minigun. To deal with the spawn overflow, you'd need some kind of controlling script. So, you decide you only need to control a single particles2d node, instead of spawning multiple ones. Then, you procede to spawn the particles node and reference it to the original 'minigun.gd', because that's were all the gun code are handled, and it'll be much easier if you handle the particles there too. And, you decide that if you were to delete the node every time it stops shooting, it would just be a waste of performance. So you just stop emitting particles whenever that happens, but keep the node.

Now you're left with a particles node linked to the gun node, and once the gun node stops existing the particles node should too. It shares everything but the position. But wait, doesn't that look like your standard sub node? Yes, it does. It's just a gimmick because you don't want the position inheritance.

And since when Godot can be compared to other engines? It's rather unique in how it works. And even if it were to be compared, doesn't mean the other ones did it quite right.

**TL;DR
It's simpler and more efficient.**

**Comment by Zephilinox** [comment]
I believe you can add a plain Node between your gun and the particles, it'll cancel out the position inheritance since Node doesn't have a position.

**Comment by mateusak** [comment]
@Zephilinox WHAT?! DAMN YOU'RE GOOD SON! Give this man a medal or something.

It would still be nice to have it, but now it's more of a minor quality of life.

**Comment by Zephilinox** [comment]
I agree there should probably be more functionality for this, adding a plain Node feels quite hackish to achieve this effect. It doesn't work when you want to keep a positional inheritance but not a rotational one for example.

**Comment by mateusak** [comment]
@Zephilinox Yeah, I don't think particles should be treated as other nodes in this respect. Particles aren't like other nodes; they don't provide structural support, they're simply visual. In this respect I also think the Sprite node need to have something similar, in the case someone wants for example an animated hand-crafted explosion to simulate particles. Then they would need to use the 'Node' hack you mentioned. Heck if it's not too much work I would add it to every node2d. Wouldn't hurt would it?

Anyways, thanks, you helped me a lot.

**Comment by Zylann** [comment]
Another reason why I don't really like this specifically on `Particles` is, that you are generalizing a particular "workable" issue on a general-purpose node. Your explosion is what we call a "VFX", a visual effect, here for a kind of explosion. Not all visual effects are only made of particles, they can be made of many different nodes (animated sprites, even sound, be it in 2D or 3D), and I'm not even mentionning projectiles, which also are always spawned outside of shooter's hierarchy. I find it much more easier to just consider them outside of hierarchy rather than hacking them inside with a boolean, while you could simply reference the effect in a variable of your gun if you need control over it.

**Comment by Zephilinox** [comment]
To be clear I think it's an issue with nodes in general, not particles specifically.

**Comment by capnm** [comment]
>  More options is always better.

A shitload of pointless options in the most contexts is what makes godot, or more generally, any software for humans useless.

**Comment by trollworkout** [comment]
wait so you want to control particle position because you want the particle generator to be a child of gun but spawn particles at your chosen x y position?

why? just spawn the particles at the target location. why do the particles have to be a child of gun? seems extra complicated code for a minimal use feature

**Comment by mateusak** [comment]
@capnm Pointless options. This is not pointless.

@trollworkout I already wrote it like a dozen times... I don't know why it's so hard to believe me, but in my case the **literal** only functional difference between spawning outside or inside is the position. So why would I write a particles controller?

**Comment by trollworkout** [comment]
@mateusak  honestly i think this is a feature only you would benefit from because you like to make particles children of the objects that spawn them .

**Comment by mateusak** [comment]
@trollworkout Sure.

**Comment by Calinou** [comment]
> Implement a 'Disable position inheritance' checkbox for Particles2D nodes.

This particular feature is available in any Node2D by calling `set_as_toplevel(true)`.

___

Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
